cpp/src/rolling/rolling_detail.cuh

```diff
@@ -1063,6 +1063,81 @@ struct rolling_window_launcher {
     return gather_map;
   }
 
+  /**
+   * @brief Count null entries in result of COLLECT.
+   */
+  size_type count_child_nulls(column_view const& input,
+                              std::unique_ptr<column> const& gather_map,
+                              rmm::cuda_stream_view stream)
+  {
+    auto input_device_view = column_device_view::create(input, stream);
+
+    auto input_row_is_null = [d_input = *input_device_view] __device__(auto i) {
+      return d_input.is_null_nocheck(i);
+    };
+
+    return thrust::count_if(rmm::exec_policy(stream),
+                            gather_map->view().template begin<size_type>(),
+                            gather_map->view().template end<size_type>(),
+                            input_row_is_null);
+  }
+
+  /**
+   * @brief Purge entries for null inputs from gather_map, and adjust offsets.
+   */
+  std::pair<std::unique_ptr<column>, std::unique_ptr<column>> purge_null_entries(
+    column_view const& input,
+    std::unique_ptr<column> const& gather_map,
+    std::unique_ptr<column> const& offsets,
```

Comment:
There's almost never a reason to pass a smart pointer by `const&` as a function shouldn't care how the objects lifetime is managed. 
```suggestion
    column_view const& gather_map,
    column_view const& offsets,
```

Reply:
Agreed. This should've been obvious. :/